src/DetectableOverflow.tsx

```diff
@@ -62,19 +62,21 @@ export class DetectableOverflow extends React.Component<Props, States> {
     const tag = this.props.tag ? this.props.tag : defaultTag;
     const style = this.props.style ? this.props.style : defaultStyle;
     const className = this.props.className ? this.props.className : '';
-    const props = {
-      style,
-      className,
-      ref: (el: Element | null) => {
-        this.ref = (el instanceof HTMLElement) ? el : null;
-      },
-    };
 
-    return React.createElement(
-      tag,
-      props,
-      <ReactResizeDetector handleWidth onResize={this.updateState} />,
-      this.props.children,
+    return (
+      <ReactResizeDetector handleWidth onResize={this.updateState} targetRef={this.ref}>
+        {
+          React.createElement(
+            tag,
+            {
+              style,
+              className,
+              ref: this.ref,
+            },
+            this.props.children,
```

Comment:
Oh, I didn't thought about this solution. That's great. I don't think it will break any apps.